py3-build.yaml

```diff
@@ -31,31 +34,53 @@ pipeline:
       tag: ${{package.version}}
       expected-commit: 1e67c062e9d1b1a6d5ffed621f4b29902bb764e5
 
-  - runs: |
-      python3 -m gpep517 build-wheel --wheel-dir dist --output-fd 1
-      python3 -m installer -d "${{targets.destdir}}" dist/*.whl
-      find ${{targets.destdir}} -name "*.pyc" -exec rm -rf '{}' +
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: ${{vars.pypi-package}} installed for python${{range.key}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - py3-${{vars.pypi-package}}
+      runtime:
+        - py${{range.key}}-packaging
+        - py${{range.key}}-pyproject-hooks
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}-bin
+    description: Executable binaries for ${{vars.pypi-package}} installed for python${{range.key}}
+    dependencies:
+      runtime:
+        - py${{range.key}}-${{vars.pypi-package}}
+      provides:
+        - py3-${{vars.pypi-package}}
+      provider-priority: ${{range.value}}
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr
+          mv ${{targets.contextdir}}/../py${{range.key}}-${{vars.pypi-package}}/usr/bin ${{targets.contextdir}}/usr
 
-  - uses: strip
+  - name: py3-supported-${{vars.pypi-package}}
+    description: meta package providing ${{vars.pypi-package}} for supported python versions.
+    dependencies:
+      runtime:
+        - py3.10-${{vars.pypi-package}}
```

Comment:
Should this be depending on the -bin version which pulls in the binaries, and then transivitely the libs?

Reply:
Asking mainly because I'm trying to document the new way of doing things 🤣

Reply:
Supported is really for us to build packages and we can just use  `python -m build` vs the binary build (which I don't think is heavily used)